src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingToAndFromManyAddressesSpecification_Steps.cs

```diff
@@ -0,0 +1,189 @@
+        private IDictionary<string, CoreNode> nodes;
+        private NodeGroupBuilder nodeGroupBuilder;
+        private const string NodeOne = "one";
+        private const string NodeTwo = "two";
+        private long stratisReceiverTotalAmount;
+        private TransactionBuildContext transactionBuildContext;
+        private int CoinBaseMaturity;
+
+        // Output helper allows test step names to be logged.
+        public SendingToAndFromManyAddressesSpecification(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+        protected override void BeforeTest()
+        {
+            this.sharedSteps = new SharedSteps();
+            this.nodeGroupBuilder = new NodeGroupBuilder();
+        }
+
+        protected override void AfterTest()
+        {
+            this.nodeGroupBuilder.Dispose();
+        }
+
+        private void two_connected_nodes()
+        {
+            this.nodes = this.nodeGroupBuilder
+                .StratisPowNode(NodeOne).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .StratisPowNode(NodeTwo).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .WithConnections()
+                    .Connect(NodeOne, NodeTwo)
+                    .AndNoMoreConnections()
+                .Build();
+        }
+
+        private void node1_sends_funds_to_node2_TO_fifty_addresses()
+        {
+            // Mine block
+            this.CoinBaseMaturity = (int)this.nodes[NodeOne].FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            this.sharedSteps.MineBlocks(this.CoinBaseMaturity + 50
+                , this.nodes[NodeOne]
+                , WalletAccountName
+                , WalletName
+                , WalletPassword);
+
+            // Sync both nodes.
+            this.nodes[NodeOne].CreateRPCClient().AddNode(this.nodes[NodeTwo].Endpoint, true);
+
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[NodeTwo], this.nodes[NodeOne]);
+
+            // Get 50 unused addresses from node two.
+            IEnumerable<HdAddress> recevierAddresses = this.nodes[NodeTwo].FullNode.WalletManager()
+                .GetUnusedAddresses(new WalletAccountReference(WalletName, WalletAccountName), 50);
+
+            List<Recipient> recipients = recevierAddresses.Select(address => new Recipient
+            {
+                ScriptPubKey = address.ScriptPubKey,
+                Amount = Money.COIN
+            }).ToList();
+
+            this.transactionBuildContext = SharedSteps.CreateTransactionBuildContext(WalletName
+                , WalletAccountName
+                , WalletPassword
+                , recipients
+                , FeeType.Medium
+                , 101);
+
+            Transaction transaction = this.nodes[NodeOne].FullNode.WalletTransactionHandler().
+                BuildTransaction(this.transactionBuildContext);
+
+            transaction.Outputs.Count.Should().Be(51);
```

Comment:
why 51 and not 50?  change address?

Reply:
This was the count after building the transaction.

Reply:
You expected there to be 51 and not 50?  why?  from what I see you pass 50 recipient addresses in

Reply:
This is 50 for nodeTwo addresses and the extra output is change - confirmed this with Dan.